lib/features/family/features/profiles/screens/profile_selection_screen.dart

```diff
@@ -326,21 +331,46 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
     return gridItems;
   }
 
-  void showPreferredChurchModal() {
+  void showPreferredChurchModal(UserExt user) {
     context.read<ProfilesCubit>().setPreferredChurchModalShown();
     FunModal(
       title: 'Choose your church',
       icon: const FunIcon(
         iconData: FontAwesomeIcons.church,
-        circleColor: FamilyAppTheme.primary95,
       ),
       subtitle: "Let's link your church to make giving easier",
       buttons: [
         FunButton(
           text: 'Continue',
-          onTap: () {
-            // TODO: kids-1447 - navigate to church selection
-            context.pop(); //for now
+          onTap: () async {
+            context.pop(); // close modal
+            getIt<OrganisationBloc>().add(
```

Comment:
I think we can move all of this to the PreferredChurchSelectionPage. Which would make it easier to reuse from different places. We can huddle on it tomorrow!